aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -255,10 +255,10 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
 
   // If dst is contiguous and there is no byte offset, we can save directly the result of
   // gather into dst. This reduces the overhead of doing an additional blit for most cases
-  bool returnGatherOutput = (dst_.is_contiguous() && !dst_byte_offset);
+  bool returnGatherOutput = (dst_.is_contiguous() && !dst_byte_offset && src_.dtype() == dst_.dtype());
```

Comment:
Just curious, why `dst_byte_offset` is still a condition there? It feel wrong to guard any logic on whether or not tensor is a contiguous slice of another tensor

Reply:
This `dst_byte_offset` check was added to return early in case dst didn't have any storage offset. 
E.g in case of no storage offset, we'll save the gather result directly in dst and return early. Otherwise, if dst has a storage offset, we cannot return early since we'd need to blit the data at that offset specified by dst.